packages/core/src/domain/configuration/experimentalFeatures.ts

```diff
@@ -0,0 +1,26 @@
+let enabledExperimentalFeatures: Set<string>
+
+export function updateExperimentalFeatures(enabledFeatures: string[]): void {
+  // Safely handle external data
+  if (!Array.isArray(enabledFeatures)) {
+    return
+  }
+
+  if (!enabledExperimentalFeatures) {
+    enabledExperimentalFeatures = new Set(enabledFeatures)
+  }
+
+  enabledFeatures
+    .filter((flag) => typeof flag === 'string')
+    .forEach((flag: string) => {
+      enabledExperimentalFeatures.add(flag)
+    })
+}
+
+export function isExperimentalFeatureEnabled(featureName: string): boolean {
+  return !!enabledExperimentalFeatures && enabledExperimentalFeatures.has(featureName)
```

Comment:
no unit test for `!!enabledExperimentalFeatures`, what was your intent?

Reply:
`!!enabledExperimentalFeatures` is tested when `isExperimentalFeatureEnabled` is called after a `resetExperimentalFeatures`.